Sources/Plasma/FeatureLib/pfPython/plPythonFileMod.h

```diff
@@ -200,16 +109,153 @@ class plPythonFileMod   : public plMultiModifier
+     *          method called when the plPythonFileMod receives a plNotifyMsg.
+     */
+    void ICallScriptMethod(func_num methodId);
+
+protected:
+    friend class plPythonSDLModifier;
+
+    plPythonSDLModifier* fSDLMod;
+
+    bool IEval(double secs, float del, uint32_t dirty);
+
+    ST::string IMakeModuleName(const plSceneObject* sobj);
+
+    ST::string fPythonFile;
+    ST::string fModuleName;
+
+    /** the list of receivers that want to be notified */
+    std::vector<plKey> fReceivers;
+
+    /** pyKey of the target plSceneObject */
+    PyObject*   fSelfKey;
+    plPipeline* fPipe;
+
+    /** List of ptAttributes */
+    std::vector<plPythonParameter> fParameters;
+
+    /** Python Module */
+    PyObject*   fModule;
+
+    /** Instance of the python script class */
+    PyObject*   fInstance;
+
+    /** Set when converting in 3ds Max indicating to NOT run the script */
+    static bool fAtConvertTime;
+
+    /** The last plNotifyMsg was sent by the local client */
+    bool        fLocalNotify;
+
+    /**
+     * Flag to determine when the first time at the eval, so the Python coders can have
+     * a chance to run initialization code after the system is up, but before things are displayed
+     */
+    bool        fIsFirstTimeEval;
+
+    /** This python script is attached to a cloned key */
+    bool        fAmIAttachedToClone;
+
+    // callback class for the KI
+    PythonVaultCallback* fVaultCallback;
+    pfPythonKeyCatcher * fKeyCatcher;
+
+    struct NamedComponent
+    {
+        ST::string  name;
+        int32_t     id;
+        bool        isActivator;
+
+        NamedComponent()
+            : name(), id(), isActivator()
+        { }
+
+        NamedComponent(const ST::string& n, int32_t i, bool a)
+            : name(n), id(i), isActivator(a)
+        { }
+    };
+
+    std::vector<NamedComponent> fNamedCompQueue;
+
+    void IFindResponderAndAdd(const ST::string& responderName, int32_t id);
+    void IFindActivatorAndAdd(const ST::string& activatorName, int32_t id);
+    void ISetKeyValue(const plKey& key, int32_t id);
+
+    bool ILoadPythonCode();
+
+    enum genref_whats
+    {
+        kNotSure = 0,
+        kAddNotify
+    };
+
+public:
+
+    plPythonFileMod();
+    ~plPythonFileMod();
+
+    CLASSNAME_REGISTER(plPythonFileMod );
+    GETINTERFACE_ANY(plPythonFileMod, plMultiModifier);
+
+    plPythonSDLModifier* GetSDLMod() const { return fSDLMod; }
+    bool WasLocalNotify() const { return fLocalNotify; }
+    plPipeline* GetPipeline() const { return fPipe; }
+    void SetSourceFile(const ST::string& filename) { fPythonFile = filename; }
+    int getPythonOutput(std::string* line);
+    void ReportError();
+    void DisplayPythonOutput();
+    static void SetAtConvertTime() { fAtConvertTime = true; }
+    bool AmIAttachedToClone() const { return fAmIAttachedToClone; }
+
+    void AddToNotifyList(const plKey& pKey) { fReceivers.push_back(pKey); }
```

Comment:
Some types of copies can be avoided with:
`void AddToNotifyList(plKey key) { fReceivers.emplace_back(std::move(pKey)); }`